src/coreclr/vm/callhelpers.cpp

```diff
@@ -471,7 +471,11 @@ void MethodDescCallSite::CallTargetWorker(const ARG_SLOT *pArguments, ARG_SLOT *
                     case 1:
                     case 2:
                     case 4:
+#if defined(TARGET_LOONGARCH64)
+                        *((INT64*)pDest) = (INT32)pArguments[arg];
```

Comment:
This change is not right now that we have established in https://github.com/loongson/LoongArch-Documentation/pull/32 that signed integers should be sign-extended and unsigned integers should be zero-extended.

Reply:
https://github.com/loongson/LoongArch-Documentation/pull/32#discussion_r772267496

Reply:
The comment you have quoted talks about one specific example. The specific example has signed integer and so value is sign-extended.

The general rule earlier in the current version of the ABI doc is:

When passed in registers or on the stack, the unsigned integer scalars narrower than GRLEN bits are zero-extended to GRLEN bits, and the signed integer scalars are sign-extended.

https://github.com/loongson/LoongArch-Documentation/pull/32/files#diff-711b3e7b6a005b492898ac6d93f2d8d37c00e0831e210993a6f9dbb26c043717R711

This rule has to be applied here and in other places.

Reply:
> The comment you have quoted talks about one specific example. The specific example has signed integer and so the value is sign-extended.
> 
> The general rule in the current version of the ABI doc is:
> 
> > When passed in registers or on the stack, the unsigned integer scalars narrower than GRLEN bits are zero-extended to GRLEN bits, and the signed integer scalars are sign-extended.
> 
> https://github.com/loongson/LoongArch-Documentation/pull/32/files#diff-711b3e7b6a005b492898ac6d93f2d8d37c00e0831e210993a6f9dbb26c043717R711
> 
> This rule has to be applied here and in other places.

Thanks, I agree with you.
It's better to fit the signed and unsigned.
I will amend it.